htdocs/categories/class/api_categories.class.php

```diff
@@ -265,6 +265,42 @@ public function delete($id)
         );
     }
 
+    /**
+     * List categories of an object
+     *
+     * Get the list of categories linked to an object
+     *
+     * @param int       $id         Object ID
+     * @param string	$type		Type of category ('member', 'customer', 'supplier', 'product', 'contact')
+     * @param string	$sortfield	Sort field
+     * @param string	$sortorder	Sort order
+     * @param int		$limit		Limit for list
+     * @param int		$page		Page number
+     * @return array                Array of category objects
+     *
+     * @throws RestException
+     *
+     * @url GET /object/{type}/{id}
+     */
+    public function getListForObject($id, $type = 'customer', $sortfield = "s.rowid", $sortorder = 'ASC', $limit = 0, $page = 0)
```

Comment:
by default `Categorie::TYPE_PRODUCT` while other are not implemented in your code?

Reply:
or add 
```
if ($type == Categorie::TYPE_CUSTOMER && ! DolibarrApiAccess::$user->rights->societe->lire) {
            throw new RestException(401);
 }

```

Reply:
```
if ($type == Categorie::TYPE_CONTACT && ! DolibarrApiAccess::$user->rights->contact->lire) {
            throw new RestException(401);
 }
```

Reply:
Because i need product, and i would like it in v11. I need to do some research work for others right access.

Reply:
cool thank you

Reply:
do you know right access fo members and suppliers?

Reply:
fournisseur->lire
adherent->lire

Reply:
thank you. i done the modification. All types are implemented.